packages/sdk/src/identity/Identity.ts

```diff
@@ -0,0 +1,23 @@
+import { UserID, UserIDRaw } from '@streamr/utils'
+import { AbstractSigner, Provider } from 'ethers'
+import { RpcProviderSource } from '../RpcProviderSource'
+import { SignatureType } from '@streamr/trackerless-network'
+
+export const IdentityInjectionToken = Symbol('Identity')
+
+/**
+ * The {@link https://docs.ethers.org/v6/api/providers/abstract-signer/#AbstractSigner AbstractSigner} type is from the `ethers` library.
+ */
+export type SignerWithProvider = AbstractSigner<Provider>
+
+/**
+ * Identity represents a digital identity, specified by cryptographic keys,
+ * and capable of signing. You can configure new Identities in identityConfig.ts.
+ */
+export abstract class Identity {
+    abstract getUserIdString(): Promise<UserID>
+    abstract getUserIdBytes(): Promise<UserIDRaw>
```

Comment:
Could match the method names to their return types. Great to have both, and one without the suffix is a good default for typical use cases. 

```suggestion
    abstract getUserId(): Promise<UserID>
    abstract getUserIdRaw(): Promise<UserIDRaw>
```

Reply:
That's fine yeah. Perhaps we could even reconsider the name `UserID` at some point, but that's way out of scope for this PR.